api/v1alpha6/conversion.go

```diff
@@ -86,6 +86,11 @@ func restorev1alpha6MachineSpec(previous *OpenStackMachineSpec, dst *OpenStackMa
 	// Subnet is removed from v1alpha7 with no replacement, so can't be
 	// losslessly converted. Restore the previously stored value on down-conversion.
 	dst.Subnet = previous.Subnet
+	// SecurityGroups is removed from v1alpha7 with no replacement.
+	// Restore the previously stored value on down-conversion.
+	for i := range dst.Ports {
+		dst.Ports[i].SecurityGroups = previous.Ports[i].SecurityGroups
```

Comment:
The fuzzer tests don't seem to like this, btw. Didn't we decided that as ports are immutable it'd be easier just to replace the whole ports slice?

Reply:
Yes, true! I will put this back in WIP mode until I have time to fix this